### DISCUSSION TITLE
开发者例会议题收集（20221018）

### DISCUSSION THREAD

**Discussion by chuntaojun** [polarismesh/polaris#753] [Meeting]
# 开发者例会议题收集（20221018）

## 会议信息

andrewshan(单家骏) 邀请您参加腾讯会议
会议主题：Polaris社区开发者双周会
会议时间：2022/09/20 19:30-20:00 (GMT+08:00) 中国标准时间 - 北京
重复周期：2022/07/12-2022/10/04 19:30-20:00, 每两周 (周二)

点击链接入会，或添加至会议列表：
https://meeting.tencent.com/dm/kBemx9IzxmGE

#腾讯会议：701-1815-5065

复制该信息，打开手机腾讯会议即可参与

**Comment by chuntaojun** [comment]
北极星插件的开发规范及开发指南

- 北极星当前的插件架构设计
  - 分为核心模块插件以及旁路功能插件
  - 与服务端同进程运行，插件定义基于 go 的 interface
  - 插件的注册依赖于 go 的 init 机制
  - 每个插件的名称全局唯一
  - 每个插件都必须是一个单例的，需要插件自行保证并发场景下的安全性
  - 插件需要和服务端一同编译
- 存在的问题
  - 第三方插件代码的稳定性需要贡献者保证，插件更新不及时可能会影响北极星主体程序的稳定
  - 主仓库代码维护的工作量变大
  - 插件无法自己控制版本更新
  - 用户用不上这些插件，但是正式 release 版本会连带这些第三方插件一并构建编译
- 后期计划的优化措施
  - 第三方插件单独项目组织进行维护，方便插件独立进行版本迭代
  - https://github.com/polarismesh/polaris/issues/671

**Reply by chuntaojun** [reply]
分插件类型

- 核心插件建议同进程
- 旁路功能插件可以考虑看看 hashicorp/go-plugin，即可以考虑跨进程

**Reply by chuntaojun** [reply]
待设计

**Comment by bear0408** [comment]
插件支持自定义扩展，配置文件可以配置脱敏处理，或者代码动态脱敏处理

**Reply by chuntaojun** [reply]
配置中心支持配置加密在规划中，这里需要给一下期望加密的粒度

**Comment by chuntaojun** [comment]
社区同学 @MI-cool 希望重新讨论的几个 issue

- https://github.com/polarismesh/polaris/issues/736
- https://github.com/polarismesh/polaris/issues/668
- https://github.com/polarismesh/polaris/issues/703
- https://github.com/polarismesh/polaris/issues/614

**Comment by chuntaojun** [comment]
北极星迭代 1.13.0 版本规划

- https://github.com/polarismesh/polaris/milestone/13

**Comment by chuntaojun** [comment]
## 北极星开发者周会 20221018 小结

- 针对议题：北极星插件的开发规范及开发指南
  - 结论
    - 插件分为核心链路插件以及旁路功能插件
    - 核心链路插件的编译需要和北极星服务端源码一起编译，保证同进程运行
    - 旁路功能插件，社区同学提议可以看看 hashicorp/go-plugin，即跨进程插件，这里社区对此建议认可并采纳，并单独开启了issue进行讨论设计：https://github.com/polarismesh/polaris/issues/757

- 针对议题：插件支持自定义扩展，配置文件可以配置脱敏处理，或者代码动态脱敏处理
  - 结论
    - 北极星配置中心的能力规划中，包含了配置文件加密
    - 配置文件加密功能目前还在规划中，暂未开始设计，预计下一两个迭代开始

- 针对议题：社区同学 @MI-cool 希望重新讨论的几个关于北极星与eureka兼容的问题
  - 结论
    - 社区接纳对于北极星支持eureka同步协议的提议，并且该功能会放在eurekaserver的apiserver插件中实现
    - 社区接纳对于支持 serviceid 大小写不敏感的提议，不过分两个阶段处理
      - 一阶段：仅支持针对 eureka 注册的服务，对服务名称的大小写设计一个开关，该开关决定 eureka 注册的服务名是按照全大小保存还是全小写保存
      - 二阶段：北极星本身支持大小写不敏感开关配置，开启之后，北极星所有服务名称均按照小写存储

- 针对议题：北极星的sidecar + envoy 无法使用北极星的限流能力
  - 结论
    - 该功能社区在规划设计中，预期按照两个阶段
      - 一阶段：先支持 envoy 的 LocalRateLimit 以及 在 polaris-sidecar 中实现RLS 协议，提供单机限流以及分布式限流能力
      - 二阶段：支持 envoy 的 filter 机制，将北极星的限流全功能以 envoy 插件的能力进行提供
